app/graphql/types/query_type.rb

```diff
@@ -13,13 +13,25 @@ def market(id:)
       Market.find(id)
     end
     # Return markets by location
-    field :markets_by_location, [Types::MarketType], null: false do
+    field :markets_by_location, Types::ReturnType, null: false do
       argument :lat, Float, required: true
       argument :lng, Float, required: true
       argument :radius, Integer, required: true
+      argument :products, [GraphQL::Types::String], required: false
     end
-    def markets_by_location(lat:, lng:, radius:)
-      Market.near([lat, lng], radius)
+    def markets_by_location(lat:, lng:, radius:, products: '')
+      markets = Market.near([lat, lng], radius)
+      if !products.empty?
+        markets = markets.joins(:products)
```

Comment:
I wonder if eventually there would be a way to abstract away some of the behavior of this method, think it's fine for now though

Reply:
what a beast of a query haha